app/forms/form.rb

```diff
@@ -60,4 +62,32 @@ def page_sequence
       params[:slug]
     )
   end
+
+  def editable_attribute_names
+    Claim::PermittedParameters.new(claim).flatten_keys & attribute_names.map(&:to_sym)
+  end
+
+  def attributes_with_current_value
+    attributes.each_with_object({}) do |(attribute, _), attributes|
+      attributes[attribute] = permitted_params[attribute]
+      next unless attributes[attribute].nil?
+
+      attributes[attribute] = load_current_value(attribute)
+    end
+  end
+
+  def load_current_value(attribute)
+    # TODO: re-implement when the underlying claim and eligibility data sources
+    # are moved to an alternative place e.g. a session hash
+
+    # Some, but not all attributes are present directly on the claim record.
+    return claim.public_send(attribute) if claim.has_attribute?(attribute)
```

Comment:
Currently thinking about this in the `PersonalDetailsForm` - there are attributes on the form (`day`, `month`, `year` which don't map directly to an attribute on the claim (they need to be derived from `date_of_birth`).

As it stands, this method would return something unexpected if no attribute is found on either the `claim` or its `eligibility` - an array of claims!

Reply:
@AbigailMcP just realised the same as I started working on `eligibility-confirmed`, so I moved this back to draft; will think about a way around it

Reply:
Quick thought, there should be a `nil` after the loop, I must have lost it in one of the many iterations

Reply:
Don't make it a blocker for your branch, I can always bring in your changes in and deal with it if this gets stuck for a bit longer – appreciate the heads up, thanks! 🙇

Reply:
no worries! not a blocker for mine, i'm also mulling things over and this approach is really useful to see.

Reply:
I should have addressed it now – I was also considering making the pre-filling "optional", and for non-standard use cases we can always keep overriding `initialize`